### PR TITLE
[incubator/cassandra] avoid backup attempt of backup pod

### PR DESCRIPTION
**Pull request by maorfr**
Signed-off-by: Maor Friedman <maorfr@gmail.com>

#### What this PR does / why we need it:
Update backup labels and label selector to avoid backup attempt of backup pod

#### Special notes for your reviewer:
Thanks!

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped

### CODE REVIEW

File: incubator/cassandra/templates/backup/cronjob.yaml [hunk 1]
```diff
@@ -35,7 +35,7 @@ spec:
             - --namespace
             - {{ $release.Namespace }}
             - --selector
-            - release={{ $release.Name }}
+            - app={{ template "cassandra.name" $ }}
```
Comment (maver1ck) [inline incubator/cassandra/templates/backup/cronjob.yaml]: Are you sure this shouldn't be equal to line 12 ?

**Review by maver1ck — CHANGES_REQUESTED** [review]

File: incubator/cassandra/templates/backup/cronjob.yaml (hunk 1 above)
Comment (maorfr) [reply incubator/cassandra/templates/backup/cronjob.yaml]: Should be the same as the StatefulSet app label, and different from the CronJobs app label

Comment (maver1ck) [reply incubator/cassandra/templates/backup/cronjob.yaml]: OK. What we have 2 releases in one namespace ? 
Should we leave release={{ $release.Name }} part ?

Comment (maorfr) [reply incubator/cassandra/templates/backup/cronjob.yaml]: You are right. included both :)

**Pull request merged**